file/writable_file_writer.cc

```diff
@@ -237,6 +240,9 @@ IOStatus WritableFileWriter::Close() {
           auto finish_ts = FileOperationInfo::FinishNow();
           NotifyOnFileSyncFinish(start_ts, finish_ts, s,
                                  FileOperationType::kFsync);
+          if (!interim.ok()) {
+            NotifyOnIOError(interim, "Fsync", file_name());
+          }
```

Comment:
Would the code end up being simpler if in all of the NotifyOnFileXXX methods, if the input status was an error, you called NotifyOnIOError?  There seems like places now that are missing the NotifyOnIOError (cases in Flush() and Append()) ?

Reply:
I have added the error notification when `writable_file_->` calls the underlying File system APIs.  I will update the PR to specify this. So in Flush(), it will report error when it calls `writable_file_->Flush`.